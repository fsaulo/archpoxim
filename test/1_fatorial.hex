0xDC000011
0x00000000
0x00000000
0x00000000
0x00000000
0x00000000
0x00000000
0x00000000
0x5C010000
0xD0000002
0x00400001
0xDC000005
0x28000040
0x4C210001
0xE7FFFFF9
0x2C000040
0x10420800
0x7C000000
0x03C07FFC
0x00200003
0xE7FFFFF3
0xFC000000
