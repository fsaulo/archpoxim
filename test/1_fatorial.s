.text
	// Inicializacao
	init:
		bun main
		.align 5
	// Funcao fatorial
	fatorial:
		// Caso base
		cmpi r1, 0
		bne 2
		mov r2, 1
		bun 5
		// Caso recursivo
		push r1
		subi r1, r1, 1
		call fatorial
		pop r1
		mul r2, r2, r1
		// Retorno da funcao
		ret
	// Funcao principal
	main:
		// SP = 32 KiB
		mov sp, 0x7FFC
		// fatorial(3)
		mov r1, 3
		call fatorial
		// Fim
		int 0
