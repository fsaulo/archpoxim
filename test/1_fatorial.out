[START OF SIMULATION]
0x00000000:	bun 17                   	PC=0x00000048
0x00000048:	mov sp,32764             	SP=0x00007FFC
0x0000004C:	mov r1,3                 	R1=0x00000003
0x00000050:	call -13                 	PC=0x00000020,MEM[0x00007FFC]=0x00000054
0x00000020:	cmpi r1,0                	SR=0x00000000
0x00000024:	bne 2                    	PC=0x00000030
0x00000030:	push r1                  	MEM[0x00007FF8]{0x00000003}={R1}
0x00000034:	subi r1,r1,1             	R1=R1-0x00000001=0x00000002,SR=0x00000000
0x00000038:	call -7                  	PC=0x00000020,MEM[0x00007FF4]=0x0000003C
0x00000020:	cmpi r1,0                	SR=0x00000000
0x00000024:	bne 2                    	PC=0x00000030
0x00000030:	push r1                  	MEM[0x00007FF0]{0x00000002}={R1}
0x00000034:	subi r1,r1,1             	R1=R1-0x00000001=0x00000001,SR=0x00000000
0x00000038:	call -7                  	PC=0x00000020,MEM[0x00007FEC]=0x0000003C
0x00000020:	cmpi r1,0                	SR=0x00000000
0x00000024:	bne 2                    	PC=0x00000030
0x00000030:	push r1                  	MEM[0x00007FE8]{0x00000001}={R1}
0x00000034:	subi r1,r1,1             	R1=R1-0x00000001=0x00000000,SR=0x00000040
0x00000038:	call -7                  	PC=0x00000020,MEM[0x00007FE4]=0x0000003C
0x00000020:	cmpi r1,0                	SR=0x00000040
0x00000024:	bne 2                    	PC=0x00000028
0x00000028:	mov r2,1                 	R2=0x00000001
0x0000002C:	bun 5                    	PC=0x00000044
0x00000044:	ret                      	PC=MEM[0x00007FE4]=0x0000003C
0x0000003C:	pop r1                   	{R1}=MEM[0x00007FE4]{0x00000001}
0x00000040:	mul r0,r2,r2,r1          	R0:R2=R2*R1=0x0000000000000001,SR=0x00000000
0x00000044:	ret                      	PC=MEM[0x00007FEC]=0x0000003C
0x0000003C:	pop r1                   	{R1}=MEM[0x00007FEC]{0x00000002}
0x00000040:	mul r0,r2,r2,r1          	R0:R2=R2*R1=0x0000000000000002,SR=0x00000000
0x00000044:	ret                      	PC=MEM[0x00007FF4]=0x0000003C
0x0000003C:	pop r1                   	{R1}=MEM[0x00007FF4]{0x00000003}
0x00000040:	mul r0,r2,r2,r1          	R0:R2=R2*R1=0x0000000000000006,SR=0x00000000
0x00000044:	ret                      	PC=MEM[0x00007FFC]=0x00000054
0x00000054:	int 0                    	CR=0x00000000,PC=0x00000000
[END OF SIMULATION]
